### ISSUE TITLE
Wireless sensors

### ISSUE THREAD

**Issue by josephluck** [josephluck/urban-jungle#2]
## TLDR

Wireless sensors hooked up to Urban Jungle to inform automatic care reminders.

## Requirements:

- Single powered WiFi hub
- WiFi hub connects with a household in the cloud
- WiFi hub will push data out to the cloud with sensor readings
- Cloud processes sensor data to kick off care reminders
- WiFi hub should only really be concerned with connection / data brokering of sensors, it shouldn't do much on the device itself, most of the heavy lifting needs to be done on the cloud as to minimise the need to update the code on the hub / sensors
- WiFi hub connects to small battery powered sensors, sensors will have to last a looooooong time
- Each sensor gets designated a name (probably of the plant)
- Plant view also includes connected sensors 
- Simple pairing of sensors to Wifi Hub. The app needs to be able to pick up that a new device has been registered and the user can choose which plant the device corresponds to.
- How to deal with remote code updates to the hub? For example, a new sensor type is released and the hub needs updating to be able to connect to it
- How to deal with a sensor running out of charge, re-pairing should be easy
- Show sensor battery level in the app
- User should be able to modify the plant that the sensor is attached to
- Removing a plant should remove it's associated sensor
- Don't discount the possibility of multiple sensor products, the hub should be able to be updated if new sensor products or updated versions of sensors are added later

## Types of sensor

- Soil moisture (watering)
- Soil PH (feed / repotting)
- Air quality (tbd...)
- Air humidity (mist)
- Light level (too bright / dim for plant?)

## Products:

- Arduino pro mini (super tiny Arduino):
https://store.arduino.cc/arduino-pro-mini

- Low power Arduino pro mini compatible (bit expensive though):
https://www.rocketscream.com/blog/product/mini-ultra-lorawan/

- Suuuuuuper tiny arduino compatible board (also looks compatible with NRF24L01):
https://www.seeedstudio.com/Seeeduino-XIAO-Arduino-Microcontroller-SAMD21-Cortex-M0+-p-4426.html?utm_source=blog&utm_medium=blog
also works with rocketstream's low power library https://github.com/rocketscream/Low-Power

- Low power library for Arduino pro mini:
https://github.com/rocketscream/Low-Power

- NRF24L01 Arduino radio frequency network chip:
https://lastminuteengineers.com/nrf24l01-arduino-wireless-communication/

- WiFi enabled arduino:
https://store.arduino.cc/arduino-nano-33-iot-with-headers

- WiFi enabled small device that can be used to hook up soil data over-the-air:
https://store.particle.io/products/argon

- Seeed grove sensors / system:
https://www.seeedstudio.com/Grove-Air-Quality-Sensor-v1-3-Arduino-Compatible.html

- Lithium batteries Simplisafe use:
https://www.batterystation.co.uk/battery-types/specialist-batteries/cr123a-batteries.html

- S9V11F5 buck/boost converter (takes variable battery power and outputs regulated, good for batteries):
https://www.active-robots.com/pololu-5v-step-up-step-down-voltage-regulator-s9v11f5.html

## Info:

- Johnny Five framework

- Arduino IoT Cloud:
https://www.arduino.cc/en/IoT/HomePage

- Some arduinos run code on a client (pc) and communicate with it over serial / WiFi. This is no good.

- Working with Arduino Pro Mini + FTDI interface:
https://www.youtube.com/watch?v=tKCiNepc6_4

- Arduino with remote sensors: 
https://forum.arduino.cc/index.php?topic=442150.0

- NRF24L01 auto pairing:
https://forum.arduino.cc/index.php?topic=564089.0

- Arduino + Firebase library (only certain arduinos supported):
- https://github.com/mobizt/Firebase-Arduino-WiFiNINA

- Wireless sensor mesh network with Arduino:
https://youtu.be/RrmpN1_Py_4

- Wireless network of Arduinos:
https://youtu.be/KoJ27BQdYxE

- Hooking up a soil sensor to ThingsSpeak:
https://uk.mathworks.com/help/thingspeak/MoistureMonitor.html

- Creating soil moisture sensors (no IoT / wireless portion)
https://learn.sparkfun.com/tutorials/soil-moisture-sensor-hookup-guide
https://www.instructables.com/id/Thirsty-Flamingo/

- Arduino mesh network: 
https://www.instructables.com/id/Simple-Arduino-Wireless-Mesh/

- Arduino wireless network communication:
https://www.youtube.com/watch?v=xb7psLhKTMA

- Wireless arduino sensors:
https://maniacbug.wordpress.com/2011/10/19/sensor-node/

- Taking Arduino to production:
https://www.postscapes.com/arduino-prototyping-to-product/
